lib/internal/process/promises.js

```diff
@@ -1,12 +1,33 @@
 'use strict';
 
 const promiseRejectEvent = process._promiseRejectEvent;
-const hasBeenNotifiedProperty = new WeakMap();
-const promiseToGuidProperty = new WeakMap();
-const pendingUnhandledRejections = [];
+
+const kNotified = Symbol('node-unhandledrejection-notified');
+const kGuid = Symbol('node-unhandledrejection-uid');
 let lastPromiseId = 1;
 
-exports.setup = setupPromises;
+class RejectedList {
```

Comment:
Why not simply `class RejectedList extends Array { push(promise, reason) {super.push({promise, reason})} }`?